org.bndtools.templating.gitrepo/src/org/bndtools/templating/jgit/Cache.java

```diff
@@ -10,14 +10,21 @@
 import aQute.bnd.service.url.TaggedData;
 import aQute.lib.io.IO;
 import aQute.libg.tuple.Pair;
+import bndtools.central.Central;
 
 public class Cache {
 
     private final ConcurrentMap<URI, Pair<String, byte[]>> cache = new ConcurrentHashMap<>();
 
-    public byte[] download(URI uri) throws IOException {
+	public byte[] download(URI uri) throws Exception {
         byte[] data;
-        try (HttpClient client = new HttpClient()) {
+		try {
+			HttpClient client = Central.getWorkspace()
+				.getPlugin(HttpClient.class);
+
+			if (client == null)
+				client = new HttpClient();
```

Comment:
You should close this HttpClient. The old code does this.

Reply:
I've removed it, it must be there always